flink-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JdbcDmlOptions.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.api.java.io.jdbc;
+
+import org.apache.flink.api.java.io.jdbc.dialect.JDBCDialect;
+import org.apache.flink.util.Preconditions;
+
+import javax.annotation.Nullable;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+/**
+ * JDBC sink DML options.
+ */
+public class JdbcDmlOptions extends JdbcTypedQueryOptions {
+
+	private static final long serialVersionUID = 1L;
+
+	private final String[] fieldNames;
+	@Nullable
+	private final String[] keyFields;
+	private final String tableName;
+	private final JDBCDialect dialect;
+
+	public static JdbcDmlOptionsBuilder builder() {
+		return new JdbcDmlOptionsBuilder();
+	}
+
+	private JdbcDmlOptions(String tableName, JDBCDialect dialect, String[] fieldNames, int[] fieldTypes, String[] keyFields) {
+		super(fieldTypes);
+		this.tableName = Preconditions.checkNotNull(tableName, "table is empty");
+		this.dialect = Preconditions.checkNotNull(dialect, "dialect name is empty");
+		this.fieldNames = Preconditions.checkNotNull(fieldNames, "field names is empty");
+		this.keyFields = keyFields;
+	}
+
+	public String getTableName() {
+		return tableName;
+	}
+
+	public JDBCDialect getDialect() {
+		return dialect;
+	}
+
+	public String[] getFieldNames() {
+		return fieldNames;
+	}
+
+	public Optional<String[]> getKeyFields() {
+		return Optional.ofNullable(keyFields);
+	}
+
+	/**
+	 * JDBCUpsertOptionsBuilder.
+	 */
+	public static class JdbcDmlOptionsBuilder extends JDBCUpdateQueryOptionsBuilder<JdbcDmlOptionsBuilder> {
+		private String tableName;
+		private String[] fieldNames;
+		private String[] keyFields;
+		private JDBCDialect dialect;
+		private JDBCDialect customDialect;
```

Comment:
This seems to not be used anywhere.

Reply:
Leftover from some previous iteration. 
Removed, thanks.